src/packageRequester.ts

```diff
@@ -207,6 +210,7 @@ async function resolveAndFetch (
           manifest: pkg,
           normalizedPref,
           resolution: resolution as DirectoryResolution,
+          sideEffectsCache: options.sideEffectsCache ? new Map() : await getSideEffectsCache(ctx.storePath, id),
```

Comment:
Better to call the returned property `sideEffectsCacheByEngine` or `cacheByEngine`